raft/src/main/java/org/apache/kafka/raft/internals/BatchAccumulator.java

```diff
@@ -298,28 +342,30 @@ public void close() {
         public final List<T> records;
         public final MemoryRecords data;
         private final MemoryPool pool;
-        private final ByteBuffer buffer;
+        // Buffer that was allocated by the MemoryPool (pool). This may not be the buffer used in
+        // the MemoryRecords (data) object.
+        private final ByteBuffer pooledBuffer;
 
         private CompletedBatch(
             long baseOffset,
             List<T> records,
             MemoryRecords data,
             MemoryPool pool,
-            ByteBuffer buffer
+            ByteBuffer pooledBuffer
```

Comment:
nit: might be worth using the same name in `BatchBuilder`. Currently we use `initialBuffer`